### PR TITLE
Enhance comment creation

### PR DESCRIPTION
**Pull request by Ramko9999**
This PR contains many untested changes. Pairing that with my lack of experience working on the FE, a plethora of bugs conceal themselves, only to reveal their identity under a different browser, different viewport dimensions and etc...

**What this PR accomplishes?**
- [1] Replaces the comment creation menu trigger from the `...` on the side of the selected start line with the right click within the highlighted region ([old](https://imgur.com/a/gK1xuw9) | [new](https://imgur.com/a/1mj6Wad))
- [2] Removes the `react-rewards` emoji animation with a custom "emoji shower" which falls right above the newly created comment's code. The animation will be triggered for all connected users when a new comment is created ([old](https://imgur.com/a/gK1xuw9) | [new](https://imgur.com/a/yG48ITd)).
- [3] Fixes bug where highlights remain over a selected comment when the next source code arrives ([old](https://imgur.com/a/AhZ0rAB) | [new](https://imgur.com/a/gQzBawv))

Todo:
- #57 
- #58

### CODE REVIEW

File: api-server/routes/ws.py [hunk 1]
```diff
@@ -285,15 +300,24 @@ async def on_pick():
...
                 add_comment_body.content,
                 add_comment_body.line_start,
                 add_comment_body.line_end,
                 add_comment_body.type,
                 username,
             )
-            comments = await get_comments(session)
-            await manager.broadcast(session.id, comments.dict())
+            comments_resp = await get_comments(session)
+            comment = list(
```
Comment (Ramko9999) [inline api-server/routes/ws.py:311]: Kinda nasty, but I will leave at this for now.

File: client/src/components/editor/Editor.tsx [hunk 2]
```diff
@@ -3,69 +3,91 @@ import "./Editor.css";
...
-    selectedLines,
-    setSelectedLines,
-    handleLineToggle,
-    isLineSelected,
-    isStartOfSelection,
-  } = useLineSelection();
-  const [addCommentMenuOpen, setAddCommentMenuOpen] = useState<boolean>(false);
+  const { selectedLines, setSelectedLines, handleLineToggle, isLineSelected } =
+    useLineSelection();
+  const [contextMenuProps, setContextMenuProps] = useState<EditorContextMenu>({
```
Comment (Ramko9999) [inline client/src/components/editor/Editor.tsx:62]: Will look to improve how the context menu is used later.

File: client/src/components/editor/animation/EmojiAnimation.tsx [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    <EmojiAnimation
+      count={count}
+      updateEmoji={updateEmoji}
+      createInitialEmoji={createInitialEmoji}
+      onFinish={onFinish}
+    />
+  );
+}
+
+export default EmojiShower;
```
Comment (rohanshiva) [inline client/src/components/editor/animation/EmojiAnimation.tsx:292]: This whole file is magic.

File: client/src/Interface.ts [hunk 4]
```diff
@@ -66,7 +67,16 @@ export enum RequestType {
   DELETE_COMMENT = 3,
 }
 
-export interface ResponsePayload {
+export enum GameStateDispatchEventType {
+  WS_RESPONSE = 1,
+  ACK_NEW_COMMENT = 2,
```
Comment (rohanshiva) [inline client/src/Interface.ts:72]: What is this event?

File: client/src/components/editor/Editor.tsx [hunk 5]
```diff
@@ -86,53 +108,136 @@ function Editor({ code, addComment, focusLines }: EditorProps) {
...
+                      {lineIndex + 1}
+                    </LineNo>
+                    <LineContent
+                      id={`${lineIndex}-content`}
+                      className={
+                        isLineSelected(lineIndex) || isFocusLine(lineIndex)
+                          ? "selected-line"
+                          : "line"
+                      }
+                      onContextMenu={(event: React.MouseEvent) => {
```
Comment (rohanshiva) [inline client/src/components/editor/Editor.tsx:214]: Wow, nice! I didn't know this existed.

File: client/src/components/editor/animation/EmojiAnimation.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+    }
+
+    const elements = showerGround.getElementsByClassName("emoji");
+    for (let i = 0; i < emojiSprites.length; i++) {
+      const spriteEl = elements[i] as HTMLSpanElement;
+      const { transform, fontSize, opacity } = getEmojiStyle(emojiSprites[i]);
+      spriteEl.textContent = emojiSprites[i].emoji;
+      spriteEl.style.fontSize = fontSize;
+      spriteEl.style.transform = transform;
+      spriteEl.style.opacity = opacity;
```
Comment (rohanshiva) [inline client/src/components/editor/animation/EmojiAnimation.tsx:134]: Some nice 4131 stuff right here!

**Review by rohanshiva — APPROVED** [review]

File: client/src/Interface.ts (hunk 4 above)
Comment (Ramko9999) [reply client/src/Interface.ts:72]: Any event generated from the server or from the client that could affect the state of the game? I wouldn't mind suggestions for a better name.

File: client/src/components/editor/animation/EmojiAnimation.tsx (hunk 6 above)
Comment (Ramko9999) [reply client/src/components/editor/animation/EmojiAnimation.tsx:134]: CSS is powerful!

**Pull request merged**